### PR TITLE
DRY up constants between modules

### LINKED ISSUE #16: Share honey constants between modules
DRY up SDK and agent/custom modules, things like `x-honeycomb-team` and default honey API

### PR DESCRIPTION
**Pull request by vreynolds**
* consolidate constants
* allow local publishing without secrets
* Closes #16

### CODE REVIEW

File: agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java [hunk 1]
```diff
@@ -48,7 +48,9 @@ private static void configureEnvironment() throws Exception {
         }
 
         System.setProperty("otel.exporter.otlp.headers",
-            String.format("x-honeycomb-team=%s,x-honeycomb-dataset=%s", apiKey, dataset));
+            String.format("%s=%s,%s=%s",
+                EnvironmentConfiguration.HONEYCOMB_TEAM_HEADER, apiKey,
+                EnvironmentConfiguration.HONEYCOMB_DATASET_HEADER, dataset));
```
Comment (paulosman) [inline agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java:53]: Is case important here? We've gone from lowercase to mixed case.

File: sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java [hunk 2]
```diff
@@ -197,12 +193,12 @@ public HoneycombSdk build() {
...
             } else {
-                builder.setEndpoint(DEFAULT_ENDPOINT);
+                builder.setEndpoint(EnvironmentConfiguration.DEFAULT_HONEYCOMB_ENDPOINT);
             }
 
             SpanExporter exporter = builder
-                .addHeader(HONEYCOMB_TEAM_HEADER, apiKey)
-                .addHeader(HONEYCOMB_DATASET_HEADER, dataset)
+                .addHeader(EnvironmentConfiguration.HONEYCOMB_TEAM_HEADER, apiKey)
+                .addHeader(EnvironmentConfiguration.HONEYCOMB_DATASET_HEADER, dataset)
```
Comment (paulosman) [inline sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java:201]: Ditto re: case. Might not matter. :shrug:

**Review by paulosman — APPROVED** [review]
Approved with a question :question:

File: agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java (hunk 1 above)
Comment (vreynolds) [reply agent/src/main/java/io/honeycomb/opentelemetry/HoneycombAgent.java:53]: I did verify that it still works with mixed-case. The SDK builder was already using mixed-case, and as far as I can tell, they share the underlying mechanism: https://github.com/open-telemetry/opentelemetry-java/blob/ceb31c6caf272390c2703c81aa33357a24449cc7/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/SpanExporterConfiguration.java#L81

IDK if the Java transport does anything with the casing, or our API doesn't care about it, maybe both.

File: sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java (hunk 2 above)
Comment (vreynolds) [reply sdk/src/main/java/io/honeycomb/opentelemetry/HoneycombSdk.java:201]: Yup, doesn't appear to matter. Also, here the casing stayed consistent. I just moved the constants to common.

**Pull request merged**